java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowToJdbcUtils.java

```diff
@@ -0,0 +1,123 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.driver.jdbc.utils;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+/**
+ * Helper class to convert Arrow types to JDBC required values.
+ */
+class ArrowToJdbcUtils {
+  static boolean isSigned(ArrowType.ArrowTypeID type) {
+    switch (type) {
+      case Int:
+      case FloatingPoint:
+      case Decimal:
+      case Interval:
+      case Duration:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  static int toJdbcType(ArrowType type) {
+    switch (type.getTypeID()) {
+      case Null:
+        return Types.NULL;
+      case Struct:
+        return Types.STRUCT;
+      case List:
+      case LargeList:
+      case FixedSizeList:
+        return Types.ARRAY;
+      case Int:
+        return ((ArrowType.Int) type).getBitWidth() == 64 ? Types.BIGINT : Types.INTEGER;
+      case FloatingPoint:
+        return Types.FLOAT;
+      case Utf8:
+      case LargeUtf8:
+        return Types.VARCHAR;
+      case Binary:
+      case LargeBinary:
+        return Types.VARBINARY;
+      case FixedSizeBinary:
+        return Types.BINARY;
+      case Bool:
+        return Types.BOOLEAN;
+      case Decimal:
+        return Types.DECIMAL;
+      case Date:
+        return Types.DATE;
+      case Time:
+        return Types.TIME;
+      case Timestamp:
+      case Interval:
+      case Duration:
+        return Types.TIMESTAMP;
+      default:
+        return Types.OTHER;
+    }
+
+  }
+
+  static String getClassName(ArrowType type) {
+    switch (type.getTypeID()) {
+      case List:
+      case LargeList:
+      case FixedSizeList:
+        return ArrayList.class.getCanonicalName();
+      case Map:
+        return HashMap.class.getCanonicalName();
+      case Int:
+        return ((ArrowType.Int) type).getBitWidth() == 64 ?
+                long.class.getCanonicalName() : int.class.getCanonicalName();
+      case FloatingPoint:
+        return float.class.getCanonicalName();
+      case Utf8:
+      case LargeUtf8:
+        return String.class.getCanonicalName();
+      case Binary:
+      case LargeBinary:
+      case FixedSizeBinary:
+        return byte[].class.getCanonicalName();
+      case Bool:
+        return boolean.class.getCanonicalName();
+      case Decimal:
+        return BigDecimal.class.getCanonicalName();
+      case Date:
+        return Date.class.getCanonicalName();
+      case Time:
+        return Time.class.getCanonicalName();
+      case Timestamp:
+      case Interval:
+      case Duration:
+        return Timestamp.class.getCanonicalName();
+      default:
+        return null;
```

Comment:
Should I error instead?

Reply:
Seems like it could be better. I looked into other scenarios where switch cases have been used with types and it seems that throwing an exception in the default have been done.

Reply:
I wonder if it would make sense to return `Object.class.getCanonicalName()`? I don't fully understand how this  gets used. From what I've been reading, this value would be used to determine the class of the object you'd like to set if you call `PreparedStatement.setObject` without specifying a type. However, you can also specify the type when you make the call.

I've never used `getObject` or `setObject` in JDBC so I'm not entirely sure how to best handle this scenario.

Reply:
Additionally, FWIW, these are the only types not covered:

```
    Null(Type.Null),
    Struct(Type.Struct_),
    Union(Type.Union),
    NONE(Type.NONE);
```

Reply:
Good point. I am not entirely sure about that bit too. 

@lidavidm appreciate your thoughts.

Reply:
hmm a thought for `Struct` is that the server could send along a class name as part of the Field metadata and default to `Object.class.getCanonicalName` if not given.

Reply:
Yeah I think we should make this a property exposed by the TypeInfo catalog.

Reply:
I found that we already provide `SqlTypes.getSqlTypeIdFromArrowType` to convert an Arrow type to a JDBC type. Avatica also provides a mapping of JDBC types to Java classes. I removed the new custom logic and replaced it with:

```
    final int jdbcType = SqlTypes.getSqlTypeIdFromArrowType(arrowType);
    final String className = SqlType.valueOf(jdbcType).clazz.getCanonicalName();
```

See `BaseAvaticaParameterConverter`.